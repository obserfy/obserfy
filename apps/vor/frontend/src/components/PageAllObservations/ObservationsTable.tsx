import { t, Trans } from "@lingui/macro"
import { FC, memo, ReactNode, useState } from "react"
import { Box, Button, Card, Flex } from "theme-ui"
import { borderBottom, borderFull } from "../../border"
import dayjs, { Dayjs } from "../../dayjs"
import { exportStudentObservations } from "../../export"
import { useGetCurriculumAreas } from "../../hooks/api/useGetCurriculumAreas"
import {
  Observation,
  useGetStudentObservations,
} from "../../hooks/api/useGetStudentObservations"
import useDebounce from "../../hooks/useDebounce"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ReactComponent as CalendarIcon } from "../../icons/calendar.svg"
import { OBSERVATION_DETAILS_URL } from "../../routes"
import AlertDialog from "../AlertDialog/AlertDialog"
import DatePickerDialog from "../DatePickerDialog/DatePickerDialog"
import Icon from "../Icon/Icon"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import ObservationListItem from "../ObservationListItem/ObservationListItem"
import SearchBar from "../SearchBar/SearchBar"
import Tab from "../Tab/Tab"
import Typography from "../Typography/Typography"

export const ObservationsTable: FC<{
  studentId: string
  studentName: string
}> = ({ studentId, studentName }) => {
  const [searchTerm, setSearchTerm] = useState("")
  const [startDate, setStartDate] = useState<Dayjs>()
  const [endDate, setEndDate] = useState<Dayjs>()

  const debouncedSearchTerm = useDebounce(searchTerm, 250)

  const observations = useGetStudentObservations(
    studentId,
    debouncedSearchTerm,
    startDate,
    endDate
  )
  const areas = useGetCurriculumAreas()
  const [areaFilter, setAreaFilter] = useState(0)

  const areaFilterId =
    areaFilter > 0 ? areas.data?.[areaFilter - 1].id ?? "" : ""

  const observationsByArea = {}
  observations.data?.forEach((observation) => {
    const areaId = observation.area?.id ?? "Other"
    observationsByArea[areaId] ??= []
    observationsByArea[areaId].push(observation)
  })

  const dates = observations.data?.map(
    (observation) => observation.eventTime
  ) ?? [undefined, undefined]

  return (
    <>
      <Flex p={3} sx={{ alignItems: "flex-end" }}>
        <Typography.H5 sx={{ lineHeight: 1 }}>
          <Trans>Observations </Trans>
        </Typography.H5>

        <ExportButton
          studentId={studentId}
          search={searchTerm}
          startDate={startDate || dayjs(dates[0])}
          endDate={endDate || dayjs(dates[dates.length - 1])}
          studentName={studentName}
        />
      </Flex>

      <Card variant="responsive">
        <Flex px={3} pt={3}>
          <SearchBar
            mr={3}
            sx={{ backgroundColor: "darkSurface", height: 40 }}
            value={searchTerm}
            onChange={(e) => setSearchTerm(e.target.value)}
          />
          {observations.isSuccess && (
            <DateRangeSelector
              startDate={startDate || dayjs(dates[0])}
              endDate={endDate || dayjs(dates[dates.length - 1])}
              onStartDateChange={setStartDate}
              onEndDateChange={setEndDate}
            />
          )}
          {observations.isLoading && (
            <LoadingPlaceholder
              sx={{ width: 245, height: 40, flexShrink: 0 }}
            />
          )}
        </Flex>

        <Tab
          small
          items={["All", ...(areas.data?.map((area) => area.name) ?? [])]}
          selectedItemIdx={areaFilter}
          onTabClick={(idx) => {
            setAreaFilter(idx)
          }}
        />

        {observations.isSuccess && (
          <ObservationList
            studentId={studentId}
            observations={
              areaFilter !== 0
                ? observationsByArea[areaFilterId] ?? []
                : observations.data ?? []
            }
          />
        )}

        {observations.isLoading && !observations.data && <LoadingState />}
      </Card>
    </>
  )
}

const ExportButton: FC<{
  studentId: string
  search: string | ""
  startDate: dayjs.Dayjs
  endDate: dayjs.Dayjs
  studentName: string
}> = ({ studentId, search, startDate, endDate, studentName }) => {
  const exportDialog = useVisibilityState()

  const handleExport = async () => {
    await exportStudentObservations(
      studentId,
      startDate,
      endDate,
      search,
      studentName
    )
    exportDialog.hide()
  }

  return (
    <>
      <Button ml="auto" onClick={exportDialog.show}>
        <Trans>Export</Trans>
      </Button>
      {exportDialog.visible && (
        <AlertDialog
          title={t`Export Observations`}
          body={t`This will export all currently visible observations as a csv file, continue?`}
          onNegativeClick={exportDialog.hide}
          onPositiveClick={handleExport}
        />
      )}
    </>
  )
}

const DateRangeSelector: FC<{
  startDate: dayjs.Dayjs
  endDate: dayjs.Dayjs
  onStartDateChange: (date: Dayjs) => void
  onEndDateChange: (date: Dayjs) => void
}> = ({ startDate, endDate, onStartDateChange, onEndDateChange }) => {
  const startDateDialog = useVisibilityState()
  const endDateDialog = useVisibilityState()

  return (
    <>
      <Flex
        mr="auto"
        sx={{
          flexShrink: 0,
          alignItems: "center",
          borderRadius: "default",
          backgroundColor: "darkSurface",
          ...borderFull,
        }}
      >
        <Icon as={CalendarIcon} ml={2} sx={{ display: ["none", "block"] }} />
        <Button
          variant="text"
          color="textMediumEmphasis"
          px={2}
          m={1}
          onClick={startDateDialog.show}
        >
          {startDate.format("D MMM YYYY")}
        </Button>
        <Box mb={1}>-</Box>
        <Button
          variant="text"
          color="textMediumEmphasis"
          px={2}
          m={1}
          onClick={endDateDialog.show}
        >
          {endDate.format("D MMM YYYY")}
        </Button>
      </Flex>

      {startDateDialog.visible && (
        <DatePickerDialog
          title={t`Pick start date`}
          defaultDate={startDate}
          onDismiss={startDateDialog.hide}
          onConfirm={(date) => {
            onStartDateChange(date)
            startDateDialog.hide()
          }}
        />
      )}

      {endDateDialog.visible && (
        <DatePickerDialog
          title={t`Pick end date`}
          defaultDate={endDate}
          onDismiss={endDateDialog.hide}
          onConfirm={(date) => {
            onEndDateChange(date)
            endDateDialog.hide()
          }}
        />
      )}
    </>
  )
}

const ObservationList: FC<{
  studentId: string
  observations: Observation[]
}> = memo(({ observations, studentId }) => {
  const observationsByDate: { [key: number]: ReactNode[] } = {}

  observations.forEach((observation) => {
    const date = dayjs(observation.eventTime).startOf("day").unix()
    observationsByDate[date] ??= []
    observationsByDate[date].push(
      <ObservationListItem
        key={observation.id}
        observation={observation}
        detailsUrl={OBSERVATION_DETAILS_URL(studentId, observation.id)}
        studentId={studentId}
      />
    )
  })

  const dates = Object.keys(observationsByDate).reverse()

  return (
    <Box>
      {observations.length === 0 && (
        <Typography.Body p={3}>
          <Trans>No observations found</Trans>
        </Typography.Body>
      )}

      {dates.map((date) => {
        const dateUnix = parseInt(date, 10)
        return (
          <Box key={date}>
            <Typography.Body
              py={2}
              sx={{
                ...borderBottom,
                borderColor: "borderSolid",
                textAlign: "center",
                backgroundColor: "darkSurface",
                position: "sticky",
                top: 0,
                zIndex: 100,
              }}
            >
              {dayjs.unix(dateUnix).format("D MMMM YYYY")}
            </Typography.Body>

            {observationsByDate[dateUnix]}
          </Box>
        )
      })}
    </Box>
  )
})

const LoadingState = () => (
  <Box p={3}>
    <LoadingPlaceholder mb={3} sx={{ width: "100%", height: "2rem" }} />
    <LoadingPlaceholder mb={3} sx={{ width: "100%", height: "5rem" }} />
    <LoadingPlaceholder mb={3} sx={{ width: "100%", height: "5rem" }} />
    <LoadingPlaceholder mb={3} sx={{ width: "100%", height: "5rem" }} />
    <LoadingPlaceholder sx={{ width: "100%", height: "5rem" }} />
  </Box>
)
