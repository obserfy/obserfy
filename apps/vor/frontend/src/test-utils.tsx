/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
