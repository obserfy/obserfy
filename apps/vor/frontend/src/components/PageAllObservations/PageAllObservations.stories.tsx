import { FC } from "react"
import PageAllObservations from "./PageAllObservations"

export default {
  title: "Page/PageAllObservations",
  component: PageAllObservations,
  parameters: {
    componentSubtitle: "Just a simple PageAllObservations",
  },
}

export const Basic: FC = () => <PageAllObservations studentId="sadfasdf" />
