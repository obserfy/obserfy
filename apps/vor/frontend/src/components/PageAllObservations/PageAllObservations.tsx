import { t } from "@lingui/macro"
import { FC } from "react"
import { Box } from "theme-ui"
import { getFirstName } from "../../domain/person"
import { useGetStudent } from "../../hooks/api/useGetStudent"
import { STUDENT_OVERVIEW_URL, STUDENTS_URL } from "../../routes"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import { ObservationsTable } from "./ObservationsTable"

interface Props {
  studentId: string
}
export const PageAllObservations: FC<Props> = ({ studentId }) => {
  const { data: student } = useGetStudent(studentId)

  return (
    <Box sx={{ maxWidth: "maxWidth.lg" }} mx="auto" pb={6}>
      <TopBar
        breadcrumbs={[
          breadCrumb(t`Students`, STUDENTS_URL),
          breadCrumb(getFirstName(student), STUDENT_OVERVIEW_URL(studentId)),
          breadCrumb(t`Observations`),
        ]}
      />

      <ObservationsTable
        studentId={studentId}
        studentName={getFirstName(student)}
      />
    </Box>
  )
}

export default PageAllObservations
