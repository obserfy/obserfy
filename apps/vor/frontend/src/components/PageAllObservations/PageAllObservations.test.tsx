import { render } from "../../test-utils"
import { Basic } from "./PageAllObservations.stories"

describe("PageAllObservations", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
