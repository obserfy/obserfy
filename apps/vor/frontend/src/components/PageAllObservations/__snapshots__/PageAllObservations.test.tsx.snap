// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAllObservations should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  padding-bottom: 128px;
  max-width: 900px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-13:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-13:last-child>.splitter {
  display: none;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 24px;
  line-height: 1;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: auto;
}

.emotion-16:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-16:hover,
.emotion-16:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-16:hover:disabled,
.emotion-16:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 0;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-left: 0;
  margin-right: 0;
}

@media screen and (min-width: 40em) {
  .emotion-17 {
    border-radius: 6px;
    margin-left: 16px;
    margin-right: 16px;
  }
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 13px;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: -33px;
  margin-left: 8px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  width: 24px;
  height: 24px;
  z-index: 2;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  padding-top: 8px;
  padding-bottom: 8px;
  margin-right: 16px;
  background-color: var(--theme-ui-colors-surface);
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  padding-left: 40px;
  font-size: 14px;
  width: 100%;
  width: 100%;
  background-color: var(--theme-ui-colors-darkSurface);
  height: 40px;
}

.emotion-22:autofill,
.emotion-22:autofill:hover,
.emotion-22:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-22:autofill:first-line,
.emotion-22:autofill:hover:first-line,
.emotion-22:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-22:-webkit-autofill,
.emotion-22:-webkit-autofill:hover,
.emotion-22:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-22:-webkit-autofill:first-line,
.emotion-22:-webkit-autofill:hover:first-line,
.emotion-22:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-22:hover,
.emotion-22:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-22::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-22::-moz-placeholder {
  font-style: italic;
}

.emotion-22:-ms-input-placeholder {
  font-style: italic;
}

.emotion-22::placeholder {
  font-style: italic;
}

.emotion-22:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 245px;
  height: 40px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  overflow-x: auto;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 100%;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-26 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  padding-left: 4px;
  padding-right: 4px;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textPrimary);
  white-space: nowrap;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 10;
  line-height: 2;
  font-size: 13px;
  margin-left: 8px;
  margin-right: 8px;
}

.emotion-26:after {
  background-color: var(--theme-ui-colors-textPrimary);
  border-radius: 2px 2px 0 0;
  width: 125%;
  height: 3px;
  content: '';
  margin-top: 8px;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
}

@media screen and (min-width: 40em) {
  .emotion-26 {
    font-size: 13px;
  }
}

.emotion-27 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 1px;
  width: 100%;
  background-color: var(--theme-ui-colors-border);
  position: absolute;
  bottom: 0;
}

.emotion-28 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-29 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 2rem;
}

.emotion-30 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 5rem;
}

.emotion-33 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 5rem;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/students/overview?studentId=sadfasdf"
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/students"
          >
            <span
              class="emotion-8"
            >
              Students
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/students/overview?studentId=sadfasdf"
          >
            <span
              class="emotion-8"
            />
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-13"
          >
            Observations
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-14"
      >
        <h5
          class="emotion-15"
        >
          Observations
        </h5>
        <button
          class="emotion-16"
        >
          Export
        </button>
      </div>
      <div
        class="emotion-17"
      >
        <div
          class="emotion-18"
        >
          <label
            class="emotion-19"
          >
            <div
              class="emotion-20"
            >
              <div
                class="emotion-21"
              />
              <input
                class="emotion-22"
                name="name"
                placeholder="Search"
                value=""
              />
            </div>
          </label>
          <div
            class="emotion-23"
          />
        </div>
        <div
          class="emotion-24"
        >
          <div
            class="emotion-25"
          >
            <p
              class="emotion-26"
            >
              All
            </p>
          </div>
          <div
            class="emotion-27"
          />
        </div>
        <div
          class="emotion-28"
        >
          <div
            class="emotion-29"
          />
          <div
            class="emotion-30"
          />
          <div
            class="emotion-30"
          />
          <div
            class="emotion-30"
          />
          <div
            class="emotion-33"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
